src/common/components/paymentMethods/creditCardForm/creditCardForm.component.js

```diff
@@ -96,11 +97,10 @@ class CreditCardController {
       this.creditCardPaymentForm.expiryMonth.$validate();
     });
 
-    if(!this.paymentMethod) {
```

Comment:
I think you can keep this if statement since we will be removing the input from the DOM when we are editing. That'll prevent this extra validation from being run and the input will be undefined anyways.